Source/pfile.cpp

```diff
@@ -337,13 +337,11 @@ bool pfile_ui_save_create(_uiheroinfo *heroinfo)
 	giNumberOfLevels = gbIsHellfire ? 25 : 17;
 
 	archive.RemoveHashEntries(GetFileName);
-	strncpy(hero_names[saveNum], heroinfo->name, PLR_NAME_LEN);
-	hero_names[saveNum][PLR_NAME_LEN - 1] = '\0';
+	CopyUtf8(hero_names[saveNum], heroinfo->name, PLR_NAME_LEN);

```

Comment:
Looks like it's valid to call sizeof with a runtime argument in this case (guessing since both array dimensions are known), using the destination buffer size directly seems appropriate.
```suggestion
	CopyUtf8(hero_names[saveNum], heroinfo->name, sizeof(hero_names[saveNum]));
```